### PR TITLE
Remove mantidplot calls from Indirect

### LINKED ISSUE #32821: Remove import_mantidplot calls from indirect scripts
**Describe the outcome that is desired.**

Some of the indirect scripts are still trying to import mantidplot - we should just remove all these redundant imports.

Just look for calls of `import_mantidplot` in the code base

**Describe any solutions you are considering**

**Additional context**

**Files to update**

- [x] IndirectImport.py
- [x] IndirectBayes.py
- [x] IndirectMuscat.py
- [x] IndirectNeutron.py
- [x] BayesQuasi.py
- [x] MoIDyn.py
- [x] PoldiDataAnalysis.py

### PR DESCRIPTION
**Pull request by sf1919**
**Description of work.**
Remove outdated calls to Mantidplot and replace with calls to functions within Workbench.

**To test:**

1. Check Systems tests pass
2. Review code

Fixes #32821.

*This does not require release notes* because **it is not user facing work**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by SilkeSchomann** [comment]
@sf1919 Should mantidplot also get replaced in scripts/Inelastic/IndirectReductionCommon.py?

**Comment by sf1919** [comment]
Good shout! I missed that one.

**Comment by sf1919** [comment]
IndirectReductionCommon has a number of methods used by algorithms. As a result it is not possible to used the pcolormesh import for the whole of IndirectReductionCommon as it is a qt function and therefore causes a SimpleAPITest failure (see [e63611a](https://github.com/mantidproject/mantid/pull/33559/commits/e63611ad89d32f42bf0ac226ac61244e02a94e49)). 

This can be bypassed by importing directly into the plot_reduction method as this is not currently called by any of the algorithms. This is perhaps why the mantidplot imports were also directly in the method. Whilst this is not a long term solution it does at least resolve issue #32821 .

I have therefore raised an issue regarding this specific problem separately - see #33597

**Review by SilkeSchomann — APPROVED** [review]
Confirmed that mantidplot is not called anymore from Indirect

File: scripts/Inelastic/IndirectReductionCommon.py [hunk 1]
```diff
@@ -933,9 +933,8 @@ def plot_reduction(workspace_name, plot_type):
 
     can_plot_contour = mtd[workspace_name].getNumberHistograms() > 1
     if (plot_type == 'Contour' or plot_type == 'Both') and can_plot_contour:
-        from mantidplot import importMatrixWorkspace
-        plot_workspace = importMatrixWorkspace(workspace_name)
-        plot_workspace.plotGraph2D()
+        from mantidqt.plotting.functions import pcolormesh
```
Comment (DavidFair) [inline scripts/Inelastic/IndirectReductionCommon.py:936]: Nit: all imports should happen at the top of the file, not in-line with the code for maintainability

**Review by DavidFair — CHANGES_REQUESTED** [review]
Nice cleanup, one nit where an import needs to be moved

File: scripts/Inelastic/IndirectReductionCommon.py (hunk 1 above)
Comment (sf1919) [reply scripts/Inelastic/IndirectReductionCommon.py:936]: See comment above....

Comment (DavidFair) [reply scripts/Inelastic/IndirectReductionCommon.py:936]: I missed the previous comment whilst reviewing :facepalm:

**Review by DavidFair — APPROVED** [review]

**Pull request merged**